File: services/typescript/fastify/src/routes/hooks/log.ts
import type { onRequestAsyncHookHandler } from "fastify";
import { getLogger } from "../../logger.js";

export const logHook: onRequestAsyncHookHandler = async (request, _reply) => {
  const logger = getLogger();

  logger.debug(
    { method: request.method, path: request.url },
    "Incoming request"
  );
};

File: services/python/starlette/src/routes/middleware/log.py
import structlog
from starlette.requests import Request
from starlette.responses import Response
from starlette.middleware.base import BaseHTTPMiddleware, RequestResponseEndpoint


class LogMiddleware(BaseHTTPMiddleware):
    async def dispatch(
        self, request: Request, call_next: RequestResponseEndpoint
    ) -> Response:
        structlog.get_logger().debug(
            "Incoming request", method=request.method, path=request.url.path
        )

        return await call_next(request)
